src/main/java/org/wildfly/security/auth/server/event/RealmSuccessfulAuthenticationEvent.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2015 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.auth.server.event;
+
+import org.wildfly.security.auth.server.RealmIdentity;
+import org.wildfly.security.authz.AuthorizationIdentity;
+
+/**
+ * A realm event signifying authentication success.
+ *
+ * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
+ */
+public final class RealmSuccessfulAuthenticationEvent extends RealmAuthenticationEvent {
+    private final AuthorizationIdentity authorizationIdentity;
```

Comment:
In order to implement policies like storing rehashed passwords and password transformation, I think I should add the credential name used for authentication, as well as the credential itself, and clear password guess if one is available.